test/metabase/api/search_test.clj

```diff
@@ -812,23 +812,96 @@
         (is (= 11 (call-count)))))))
 
 (deftest snowplow-new-search-query-event-test
-  (testing "Send a snowplow event when a new global search query is made"
-    (snowplow-test/with-fake-snowplow-collector
-      (mt/user-http-request :crowberto :get 200 "search?q=test")
-      (is (=? {:data {"event"                "new_search_query"
-                      "runtime_milliseconds" pos?}
+  (testing "Send a snowplow event when a search query is triggered and context is passed"
+           (snowplow-test/with-fake-snowplow-collector
+            (mt/user-http-request :crowberto :get 200 "search?q=test" :context "global-search")
+            (is
+             (=?
+              {:data    {"event"                "new_search_query"
+                         "runtime_milliseconds" pos?
+                         "context"              "global-search"}
+               :user-id (str (mt/user->id :crowberto))}
+              (last (snowplow-test/pop-event-data-and-user-id!)))))
+           (snowplow-test/with-fake-snowplow-collector
+            (mt/user-http-request :crowberto :get 200 "search?q=test" :context "filtered-search")
+            (is
+             (=?
+              {:data    {"event"                "new_search_query"
+                         "runtime_milliseconds" pos?
+                         "context"              "filtered-search"}
+               :user-id (str (mt/user->id :crowberto))}
+              (last (snowplow-test/pop-event-data-and-user-id!))))))
+
+  (testing "Don't send a snowplow event if the search doesn't contain context"
```

Comment:
Is this actually what we want to happen? why wouldn't we want to always record a search event

Reply:
We only want to record search API calls that are triggered by the search bar or the search app. The data picker in the notebook and Metabot also trigger search, but we don't care about those

Reply:
It's basically a way for product to determine how much filters are being used, how many people are using the search bar vs the search page, etc

Reply:
Ah okay. Then IMO the code and this test description should be changed to match and document this decision. 

Using a non-nil context to decide whether to send the snowplow event is obviously an implementation detail. The real spec is iff `(contains? #{"global-search" "filtered-search"} `context`)` is true, then send a snowplow event